File: sdks/javascript/examples/register-agent.ts
/**
 * Example: Register an agent and make authenticated requests.
 *
 * Run with: npx tsx examples/register-agent.ts
 */

import { ATAPClient } from "@atap/sdk";

async function main() {
  // Create an unauthenticated client for registration
  const client = new ATAPClient({
    baseUrl: "http://localhost:8080",
  });

  // Register a new agent (server generates keypair)
  const entity = await client.entities.register("agent", {
    name: "my-agent",
  });

  console.log("Registered agent:");
  console.log(`  ID:     ${entity.id}`);
  console.log(`  DID:    ${entity.did}`);
  console.log(`  Name:   ${entity.name}`);
  console.log(`  Secret: ${entity.clientSecret}`);

  // Create an authenticated client using the returned credentials
  const authed = new ATAPClient({
    baseUrl: "http://localhost:8080",
    did: entity.did,
    privateKey: entity.privateKey!,
    clientSecret: entity.clientSecret!,
  });

  // Check DIDComm inbox
  const inbox = await authed.didcomm.inbox();
  console.log(`\nInbox has ${inbox.count} messages`);

  // Check server health
  const health = await authed.discovery.health();
  console.log(`\nServer health: ${JSON.stringify(health)}`);
}

main().catch(console.error);

File: sdks/python/examples/register_agent.py
"""Example: Register an agent and make authenticated API calls."""

from atap import ATAPClient

# Step 1: Register a new agent (no auth required)
client = ATAPClient(base_url="http://localhost:8080")

entity = client.entities.register("agent", name="my-travel-agent")
print(f"Entity ID:  {entity.id}")
print(f"DID:        {entity.did}")
print(f"Secret:     {entity.client_secret}")
print(f"Private Key:{entity.private_key}")

# Step 2: Create an authenticated client
authed = ATAPClient(
    base_url="http://localhost:8080",
    did=entity.did,
    private_key=entity.private_key,
    client_secret=entity.client_secret,
)

# Step 3: List approvals (requires atap:inbox scope)
approvals = authed.approvals.list()
print(f"\nPending approvals: {len(approvals)}")

# Step 4: Check DIDComm inbox
inbox = authed.didcomm.inbox()
print(f"Messages in inbox: {inbox.count}")

# Clean up
authed.close()
client.close()
